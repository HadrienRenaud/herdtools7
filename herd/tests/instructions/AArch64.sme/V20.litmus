AArch64 V20
{
 uint32_t x[4] = {101,102,103,104};
 uint32_t y[4] = {201,202,203,204};

 0:X0=x;
 0:X1=y;

 uint32_t out1[4];
 uint32_t out2[4];

 0:X2=out1;
 0:X3=out2;
}
P0                            ;
SMSTART                       ;
PTRUE P1.S,VL4                ;
MOV W12,WZR                   ;
LD1W {ZA0H.S[W12,1]},P1/Z,[X0];
LD1W {ZA1H.S[W12,0]},P1/Z,[X1];
MOVA Z0.S,P1/M,ZA0H.S[W12,1]  ;
MOVA Z1.S,P1/M,ZA1H.S[W12,0]  ;
ST1W {Z0.S},P1,[X2]           ;
ST1W {Z1.S},P1,[X3]           ;
SMSTOP                        ;

forall out1={101,102,103,104} /\ out2={201,202,203,204}
