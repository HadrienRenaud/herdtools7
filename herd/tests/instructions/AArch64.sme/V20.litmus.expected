Test V20 Required
States 1
out1={101,102,103,104}; out2={201,202,203,204};
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Matrix-Extensions-is-work-in-progress
Condition forall (out1={101,102,103,104} /\ out2={201,202,203,204})
Observation V20 Always 1 0
Hash=bb6eb56aaad338240d1de3408688030b

